dts/bindings/pinctrl/nuvoton,npcx-pinctrl.yaml

```diff
@@ -0,0 +1,107 @@
+  label set in your SoC's devicetree, so you can modify it like this:
+
+    &pinctrl {
+            /* your modifications go here */
+    };
+
+  A group can also specify shared pin properties common to all the specified
+  pins, such as the 'pinmux-locked' property in group 0. Here is a example to
+  lock the pinmux configurations of all selected adc channels.
+
+    &adc0_default {
+      group0 {
+        pinmux = <&alt6_adc0_sl
+                  &alt6_adc2_sl>;
+      };
+      pinmux-locked;
+    };
+
+  Here is a list of supported standard pin properties:
+    - bias-disable: Disable pull-up/down (default behavior, not required).
+    - bias-pull-down: Enable pull-down resistor.
+    - bias-pull-up: Enable pull-up resistor.
+    - drive-push-pull: Output driver is push-pull (default, not required).
+    - drive-open-drain: Output driver is open-drain.
+    - output-high: Set output state high when pin configured.
+    - output-low: Set output state low when pin configured.
+
+  Custom pin properties for npcx series are available also:
+    - pinmux-locked: Lock pinmux configuration for peripheral device
+    - pinmux-gpio: Inverse pinmux back to gpio
+    - low-voltage-input: Select low-voltage input (1p8v) for io and peripheral devices
+
+  An example for NPCX7 family, include the chip level pinctrl DTSI file in the
+  board level DTS:
+
+    #include <nuvoton/npcx/npcx7/npcx7-pinctrl.dtsi>
+
+  We want to use the I2C0_0 port of the NPCX7M6FB controller and enable the
+  internal 3.3V pull-up if its i2c frequency won't exceed 400kHz.
+
+  To change a pin's pinctrl default properties, add a reference to the
+  pin in the board's DTS file and set the properties as below:
+
+    &i2c0_0_default {
+      group0{
+        bias-pull-up; /* Enable internal pull-up for i2c0_0 */
+      };
+    };
+
+    &i2c0_0 {
+      pinctrl-0 = <&i2c0_0_default>;
+      pinctrl-names = "default";
+    }
+
+compatible: "nuvoton,npcx-pinctrl"
+
+include:
+    - name: base.yaml
+    - name: pincfg-node-group.yaml
+      child-binding:
+        child-binding:
+          property-allowlist:
+            - bias-disable
+            - bias-pull-down
+            - bias-pull-up
+            - drive-push-pull
+            - drive-open-drain
+            - output-high
+            - output-low
+
+child-binding:
+    description: NPCX pin controller pin group
+    child-binding:
+      description: |
+        NPCX pin controller pin configuration state nodes
+      properties:
+        pinmux:
+          type: phandles
+          required: false
+          description: Configurations of pinmux selection
+        group-gpios:
+          type: phandle-array
+          required: false
+          description: |
+            An array of GPIOs. Each element in the array is used for IO-pad configurations
+        periph-pupd:
+          type: array
+          required: false
+          description: Pull-up/down resistors on the corresponding peripheral devices only
```

Comment:
How is this field different from the `bias-pull-up` and `bias-pull-down` properties?

Reply:
Please refer to the first message of this PR. ```bias-pull-up``` and ```bias-pull-down``` properties are used for enabling related bias types. ```periph-pupd``` is used for indicating a mapping between peripheral devices and control reg/bit. The workflow can be separated into three phases.

1. Enable ```bias-pull-up``` in peripheral pinctrl node. in the [board DT file](https://github.com/zephyrproject-rtos/zephyr/pull/43638/files#diff-52b6222516f8967ea9256d61408c141844549786916bd7c562e7e45a30de99a3R69).
2. i2c driver will add a pinctrl_soc_pin_t structure via ```periph-pupd``` [property](https://github.com/zephyrproject-rtos/zephyr/pull/43638/files#diff-47f8123ede90ae20effd65db313e6cafc7fd8af5d68368f4a29c694837519766R20).
3. Then, pinctrl driver will set the related bit of ```PUPD_EN0/1``` [here](https://github.com/zephyrproject-rtos/zephyr/pull/43638/files#diff-ae86d58960bbbd19820ec2c2cd25fe2a4415e49717f051246f51ea64d1ce4d3aR90).

Please notice not all peripheral devices have internal pull-up functionalities. If users enable ```bias-pull-up``` of the peripheral devices, they will get a build error.

Reply:
So the `periph-pupd` property is used to map to the correct `PUPD_ENn` register and bit mask.  This mapping is fixed for the SoC type and won't ever be changed by the user - is my understanding correct?

The description text should make it clear that this is an private field for use by the NPCX drivers only and that users should not change the setting.

Reply:
> So the `periph-pupd` property is used to map to the correct `PUPD_ENn` register and bit mask. This mapping is fixed for the SoC type and won't ever be changed by the user - is my understanding correct?

Yes, you're right.

> The description text should make it clear that this is an private field for use by the NPCX drivers only and that users should not change the setting.

Will do it later.

Reply:
Done.